implementation/src/main/java/io/smallrye/config/TracedPropertiesConfigSource.java

```diff
@@ -0,0 +1,47 @@
+package io.smallrye.config;
+
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.io.LineNumberReader;
+import java.net.URL;
+import java.nio.charset.StandardCharsets;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Optional;
+
+import io.smallrye.config.common.MapBackedConfigSource;
+import io.smallrye.config.common.utils.ConfigSourceUtil;
+
+public class TracedPropertiesConfigSource extends MapBackedConfigSource implements TracedConfigSource {
+    private static final long serialVersionUID = 3969193230016229478L;
+
+    private static final String NAME_PREFIX = "TracedPropertiesConfigSource[source=";
+
+    private final URL url;
+    private final Map<String, Integer> tracedProperties = new HashMap<>();
+
+    public TracedPropertiesConfigSource(URL url) throws IOException {
+        super(NAME_PREFIX + url.toString() + "]", ConfigSourceUtil.urlToMap(url));
+        this.url = url;
+        generateTracingInfo(url);
+    }
+
+    private void generateTracingInfo(URL locationOfProperties) throws IOException {
+        try (LineNumberReader reader = new LineNumberReader(
+                new InputStreamReader(locationOfProperties.openStream(), StandardCharsets.UTF_8))) {
+
+            String line;
+            while ((line = reader.readLine()) != null) {
+                findName(line).ifPresent(name -> tracedProperties.put(name, reader.getLineNumber()));
```

Comment:
This is inherently racy... we probably don't want to get the properties two separate times to generate the total set of information.

Reply:
Again, this was just to have a quick implementation. A final one would require us to write our own Properties object so we can read, parse the value and store the line number at the same time. I'm lazy, just wanted to have an implementation going :)